.github/workflows/generate-changelog.yaml

```diff
@@ -0,0 +1,127 @@
+name: Generate changelog
+# Reusable workflow to generate changelog after release
+
+on:
+  workflow_call:
+    inputs:
+      old-tag:
+        description: Version to be replaced
+        required: true
+        type: string
+      new-tag:
+        description: Version after bump
+        required: true
+        type: string
+      gh-changelog:
+        description: Path to the Changelog.md file
+        required: true
+        type: string
+      doc-changelog:
+        description: Path to the documentation changelog (if any exists)
+        required: false
+        default: ""
+        type: string
+      config:
+        description: Path to the changelog config JSON file
+        required: false
+        default: "./.github/changelog-config.json"
+        type: string
+
+    secrets:
+      github-username:
+        description: "The GitHub username for committing the changes."
+        required: true
+      github-email:
+        description: "The GitHub email for committing the changes."
+        required: true
+      github-token:
+        description: "The GitHub token for committing the changes."
+        required: true
+
+jobs:
+  read-config-file:
+    name: Read config file
+    runs-on: ubuntu-22.04
+    outputs:
+      content: ${{ steps.read-config.outputs.content }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.repository.default_branch }}
+          # required for getting all tags
```

Comment:
You don't need all tags though, do you?

Reply:
this CI pipeline is strongly based on the quick one. since i dont have any other use case to compare it with, i am using the same values quick used.

Reply:
i thought that is usually done that way when i dealing with changelog

Reply:
Okay, we use a fetch-depth of 0 there because we also have to compute the new tag. I'd assume you can omit it for this action.